tmt/base.py

```diff
@@ -1466,24 +1494,121 @@ def show(self):
-            backup = copy.deepcopy(self._context.obj)
-            run = Run(abs_path, self._context.obj.tree, self._context)
+        for abs_path in tmt.utils.generate_runs(path, id_):
+            run = Run(abs_path, self._context.obj.tree, self._context, False)
             self.process_run(run)
-            self._context.obj = backup
+
+
+class Clean(tmt.utils.Common):
+    """ A class for cleaning up workdirs, guests or images """
+
+    def images(self):
+        """ Clean images of provision plugins """
+        self.info('Image cleanup', color='blue')
+        for method in tmt.steps.provision.ProvisionPlugin.methods():
+            method.class_.clean_images(self, self.opt('dry'))
+
+    def _matches_how(self, plan):
+        """ Check if the given plan matches options """
+        how = plan.provision.data[0]['how']
+        virtual = self.opt('virtual')
+        container = self.opt('container')
+        if virtual and how == 'virtual':
+            return True
+        if container and how == 'container':
+            return True
+        # If virtual and container are not set, do not filter provision,
+        # return True, otherwise how didn't match and return False
+        return not any([virtual, container])
+
+    def _stop_running_guests(self, run):
+        """ Stop all running guests of a run """
+        if not self.load_run(run):
+            return False
+        # Clean guests if provision is done but finish is not done
+        successful = True
+        for plan in run.plans:
+            if plan.provision.status() == 'done':
+                if plan.finish.status() != 'done':
+                    # Wake up provision to load the active guests
+                    plan.provision.wake()
+                    if not self._matches_how(plan):
+                        continue
+                    if self.opt('dry'):
+                        self.verbose(f'Would stop guests in run {run.workdir} '
+                                     f'plan {plan.name}', shift=1)
+                    else:
+                        self.verbose(f'Stopping guests in run {run.workdir} '
+                                     f'plan {plan.name}', shift=1)
+                        # Set --quiet to avoid finish logging to terminal
+                        quiet = self._context.params['quiet']
+                        self._context.params['quiet'] = True
+                        try:
+                            plan.finish.go()
+                        except tmt.utils.GeneralError as error:
+                            self._context.params['quiet'] = quiet
+                            self.warn(f'Could not stop guest in run '
+                                      f'{run.workdir}: {error}', shift=1)
+                            successful = False
+                        self._context.params['quiet'] = quiet
+        return successful
+
+    def guests(self):
+        """ Clean guests of runs """
+        self.info('Guest cleanup', color='blue')
```

Comment:
In `tmt run` we use all lowercase headings. What about using just `guests` here to give consistent feeling? The `clean` part is logged already so I'd propose to omit it.